proof/refine/RISCV64/IpcCancel_R.thy

```diff
@@ -3037,7 +3037,7 @@ lemma ct_not_in_ntfnQueue:
 
 lemma sch_act_wf_weak[elim!]:
   "sch_act_wf sa s \<Longrightarrow> weak_sch_act_wf sa s"
-  by (case_tac sa, (fastforce simp: weak_sch_act_wf_def st_tcb_at'_def obj_at'_def)+)
+  by (clarsimp simp: weak_sch_act_wf_def)
```

Comment:
Haha, that's a lot nicer than what I suggested!

Reply:
I significantly weakened `weak_sch_act_wf` as part of this PR, so it really didn't need much. I should have looked more closely when I first came across this. 

About the weakening: we've got `valid_sched_action` in AInvs that has lots of information that we can cross if we need. I really just wanted there to be a TCB at the switch to target to make CRefine easier in a few spots. We often definitely do want a TCB somewhere so we can argue about `NULL` properly, together with `no_0_obj'`